packages/massa-web3/src/experimental/accountOperation.ts

```diff
@@ -0,0 +1,133 @@
+import { Account } from './account'
+import {
+  Address,
+  Operation,
+  OperationManager,
+  OperationType,
+  OptOpDetails,
+  RollOperation,
+  TransferOperation,
+  calculateExpirePeriod,
+} from './basicElements'
+import { BlockchainClient } from './client'
+
+/**
+ * A class regrouping all the account operations.
+ */
+export class AccountOperation {
+  constructor(
+    public account: Account,
+    public client: BlockchainClient
+  ) {}
+
+  private async rollOperation(
+    amount: number,
+    opts?: OptOpDetails
+  ): Promise<string> {
+    const operation = new OperationManager(this.account.privateKey, this.client)
+    const details: RollOperation = {
+      // Todo: change with fetchMinimalFees once ready
+      fee: opts?.fee ?? 0,
+      expirePeriod: await calculateExpirePeriod(
+        this.client,
+        opts?.periodToLive
+      ),
+      type: amount > 0 ? OperationType.RollBuy : OperationType.RollSell,
+      amount: Math.abs(amount),
+    }
+
+    return operation.send(details)
+  }
+
+  /**
+   * Buys rolls.
+   *
+   * @param client - The client to connect to the desired blockchain.
+   * @param amount - The number of rolls to buy.
+   * @param opts - Optional operation details.
+   *
+   * @returns The ID of the operation.
+   * @throws If the amount of rolls is not a positive non-zero value.
+   */
+  async buyRolls(
+    client: BlockchainClient,
+    amount: number,
+    opts?: OptOpDetails
+  ): Promise<string> {
```

Comment:
why this one is returning string while the others `operation` ?

Reply:
Now, buy rolls and sells rolls return a string and tranfer an operation. We can't unify ?